src/analysis/retail/paladin/protection/modules/CastLinkNormalizer.ts

```diff
@@ -0,0 +1,187 @@
+    linkingEventId: TALENTS.BLESSED_HAMMER_TALENT.id,
+    linkingEventType: EventType.Cast,
+    forwardBufferMs: BUFFER_MS,
+    backwardBufferMs: BUFFER_MS,
+    anyTarget: true,
+    isActive: (c) => c.hasTalent(TALENTS.GRAND_CRUSADER_TALENT),
+  },
+  {
+    linkRelation: GRAND_CRUSADER_BLESSED_HAMMER_CAST,
+    reverseLinkRelation: GRAND_CRUSADER_BLESSED_HAMMER_CAST,
+    referencedEventId: SPELLS.GRAND_CRUSADER_BUFF.id,
+    referencedEventType: EventType.Cast,
+    linkingEventId: SPELLS.GRAND_CRUSADER_BUFF.id,
+    linkingEventType: [EventType.ApplyBuff, EventType.RefreshBuff],
+    forwardBufferMs: BUFFER_MS,
+    backwardBufferMs: BUFFER_MS,
+    anyTarget: true,
+  },
+];
+
+const EVENT_LINKGCP: EventLink[] = [
+  {
+    linkRelation: GRAND_CRUSADER_CAST,
+    reverseLinkRelation: GRAND_CRUSADER_CAST,
+    referencedEventId: TALENTS.GRAND_CRUSADER_TALENT.id,
+    referencedEventType: [EventType.ApplyBuff, EventType.ApplyBuff],
+    linkingEventId: TALENTS.BLESSED_HAMMER_TALENT.id,
+    linkingEventType: EventType.Cast,
+    forwardBufferMs: BUFFER_MS,
+    backwardBufferMs: BUFFER_MS,
+    anyTarget: true,
+    isActive: (c) => c.hasTalent(TALENTS.GRAND_CRUSADER_TALENT),
+  },
+  {
+    linkRelation: GRAND_CRUSADER_PARRY,
+    reverseLinkRelation: GRAND_CRUSADER_PARRY,
+    referencedEventId: SPELLS.GRAND_CRUSADER_BUFF.id,
+    referencedEventType: [EventType.ApplyBuff, EventType.ApplyBuff],
+    linkingEventId: null,
+    linkingEventType: EventType.Damage,
+    forwardBufferMs: 100,
+    backwardBufferMs: 100,
+    anyTarget: true,
+    anySource: true,
+    isActive: (c) => c.hasTalent(TALENTS.GRAND_CRUSADER_TALENT),
+    additionalCondition: (linkingEvent) =>
+      linkingEvent.type === EventType.Damage && linkingEvent.hitType === HIT_TYPES.PARRY,
+  },
+];
+
+const EVENT_LINKGCJ: EventLink[] = [
+  {
+    linkRelation: GRAND_CRUSADER_CAST,
+    reverseLinkRelation: GRAND_CRUSADER_CAST,
+    referencedEventId: SPELLS.GRAND_CRUSADER_BUFF.id,
+    referencedEventType: [EventType.ApplyBuff, EventType.RefreshBuff],
+    linkingEventId: SPELLS.JUDGMENT_CAST_PROTECTION.id,
+    linkingEventType: EventType.Damage,
+    forwardBufferMs: BUFFER_MS,
+    backwardBufferMs: BUFFER_MS,
+    anyTarget: true,
+    isActive: (c) => c.hasTalent(TALENTS.GRAND_CRUSADER_TALENT),
+  },
+  {
+    linkRelation: GRAND_CRUSADER_JUDGMENT_CRIT,
+    reverseLinkRelation: GRAND_CRUSADER_JUDGMENT_CRIT,
+    referencedEventId: SPELLS.GRAND_CRUSADER_BUFF.id,
+    referencedEventType: [EventType.ApplyBuff, EventType.RefreshBuff],
+    linkingEventId: SPELLS.JUDGMENT_CAST_PROTECTION.id,
+    linkingEventType: EventType.Damage,
+    forwardBufferMs: 100,
+    backwardBufferMs: 100,
+    anyTarget: true,
+    isActive: (c) => c.hasTalent(TALENTS.GRAND_CRUSADER_TALENT),
+    additionalCondition: (linkingEvent) =>
+      linkingEvent.type === EventType.Damage && linkingEvent.hitType === HIT_TYPES.CRIT,
```

Comment:
```suggestion
      linkingEvent.hitType === HIT_TYPES.CRIT,
```
first check is unnecessary because we are filtering by damage already.

Reply:
The type of `linkingEvent` is AnyEvent, the `linkingEvent.type === EventType.Damage` is necessary to get TypeScript to play nicely with it.

Reply:
Casting it is just a bit cleaner imo then and would remain consistent with other normalizers.